### PR TITLE
[wasm][debugger] Fix source-link test

### LINKED ISSUE #62551: [wasm][debugger][tests] `DebuggerTests.MiscTests.GetSourceUsingSourceLink` failing
```
[xUnit.net 00:00:04.28]     DebuggerTests.MiscTests.GetSourceUsingSourceLink [FAIL]
  Failed DebuggerTests.MiscTests.GetSourceUsingSourceLink [769 ms]
  Error Message:
   Failed to getScriptSource: [Result: IsOk: False, IsErr: True, Value: , Error: {
  "code": -32000,
  "message": "No script for id: dotnet://2_0"
} ]
```

Investigating this with @thaystg, we found:

1. `SourceFile.GetDataAsync` fails to fetch the source file from `SourceLinkUri: https://raw.githubusercontent.com/dotnet/runtime/9943350b43caf055588a4ec38d316c25a450f27c/src/mono/wasm/debugger/tests/debugger-test-
with-source-link/test.cs`.

2. `SourceFile` also has `file:///_/src/mono/wasm/debugger/tests/debugger-test-with-source-link/test.cs` but that doesn't resolve as the proxy looks for the path under the test directory.

3. Thays pointed out that it works fine with her PR branches, and it uses `thaystg/runtime/..` for that.
4. But I'm running the tests locally, and on helix with a branch HEAD that isn't pushed anywhere at all

If this works fine with PR branches in a user's fork
  - then we don't have to fix anything. But when the test fails, we should add a message pointing out that it might be due to the above case.
  - else we need to figure out a better way to do this test

### PR DESCRIPTION
**Pull request by thaystg**
Like this we are still testing the source-link but we don't depend anymore of the git commit id.
Fix #62551

### CODE REVIEW

File: src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs [hunk 1]
```diff
@@ -816,7 +816,7 @@ private async Task<MemoryStream> GetDataAsync(Uri uri, CancellationToken token)
             {
                 if (uri.IsFile && File.Exists(uri.LocalPath))
                 {
-                    using (FileStream file = File.Open(SourceUri.LocalPath, FileMode.Open))
+                    using (FileStream file = File.Open(uri.LocalPath, FileMode.Open))
```
Comment (radical) [inline src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs:819]: Make `GetDataAsync` static.

File: src/mono/wasm/debugger/tests/debugger-test-with-source-link/debugger-test-with-source-link.csproj [hunk 2]
```diff
@@ -2,14 +2,9 @@
...
-    <EnableSourceLink>true</EnableSourceLink>
-    <IncludeSymbols>true</IncludeSymbols>
-    <DebugType>embedded</DebugType>
-    <IncludeSymbols>true</IncludeSymbols>
+    <SourceLink>source-link.json</SourceLink>
     <DeterministicSourcePaths>true</DeterministicSourcePaths>
-    <DisableSourceLink>false</DisableSourceLink>
+    <AppOutputBase>$(MSBuildProjectDirectory)\</AppOutputBase>
+    <PathMap>$(AppOutputBase)=/fakepath/$(MSBuildProjectDirectory)</PathMap>
   </PropertyGroup>
```
Comment (radical) [inline src/mono/wasm/debugger/tests/debugger-test-with-source-link/debugger-test-with-source-link.csproj:9]: Maybe:

```suggestion
<PropertyGroup>
    <PackageId>LibraryTest</PackageId>
    <Version>1.0.0</Version>
</PropertyGroup>

<PropertyGroup Condition="'$(ContinuousIntegrationBuild)' == 'true'">
    <PublishRepositoryUrl>true</PublishRepositoryUrl>
    <RepositoryUrl>https://github.com/dotnet/runtime.git</RepositoryUrl>
    <RepositoryType>git</RepositoryType>
    <EnableSourceLink>true</EnableSourceLink>
    <IncludeSymbols>true</IncludeSymbols>
    <DebugType>embedded</DebugType>
    <IncludeSymbols>true</IncludeSymbols>
    <DeterministicSourcePaths>true</DeterministicSourcePaths>
    <DisableSourceLink>false</DisableSourceLink>
</PropertyGroup>

<PropertyGroup Condition="'$(ContinuousIntegrationBuild)' != 'true'">
    <SourceLink>source-link.json</SourceLink>
    <DeterministicSourcePaths>true</DeterministicSourcePaths>
    <AppOutputBase>$(MSBuildProjectDirectory)\</AppOutputBase>
    <PathMap>$(AppOutputBase)=/fakepath/$(MSBuildProjectDirectory)</PathMap>
</PropertyGroup>
```

File: src/mono/wasm/debugger/tests/debugger-test-with-source-link/source-link.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+    "documents": {
+        "/fakepath/*":               ""
```
Comment (radical) [inline src/mono/wasm/debugger/tests/debugger-test-with-source-link/source-link.json:3]: Does `GetDataAsync` still get called, given the empty string here?

Comment (thaystg) [reply src/mono/wasm/debugger/tests/debugger-test-with-source-link/source-link.json:3]: Yes, that is why I detected that the code there was wrongly using `SourceUri.LocalPath`

Comment (radical) [reply src/mono/wasm/debugger/tests/debugger-test-with-source-link/source-link.json:3]: What is SourceUri, and SourceLinkUri in this case?

File: src/mono/wasm/debugger/tests/debugger-test-with-source-link/debugger-test-with-source-link.csproj [hunk 4]
```diff
@@ -1,10 +1,22 @@
 <Project Sdk="Microsoft.NET.Sdk">
-  <PropertyGroup>
+<PropertyGroup>
     <PackageId>LibraryTest</PackageId>
     <Version>1.0.0</Version>
+    <DeterministicSourcePaths>true</DeterministicSourcePaths>
```
Comment (radical) [inline src/mono/wasm/debugger/tests/debugger-test-with-source-link/debugger-test-with-source-link.csproj:5]: trailing spaces

**Review by radical — APPROVED** [review]
LGTM! 👍

**Pull request merged**